grpc/dispatcher.go

```diff
@@ -0,0 +1,86 @@
+package grpc
+
+import (
+	"context"
+	"time"
+
+	"github.com/gojek/fiber"
+	fiberError "github.com/gojek/fiber/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	TimeoutDefault = time.Second
+)
+
+type Dispatcher struct {
+	Timeout time.Duration
+}
+
+func (d *Dispatcher) Do(request fiber.Request) fiber.Response {
+
+	grpcRequest, ok := request.(*Request)
+	if !ok {
+		return fiber.NewErrorResponse(
+			fiberError.FiberError{
+				Code:    int(codes.InvalidArgument),
+				Message: "fiber: grpc.Dispatcher supports only grpc.Request type of requests",
+			})
+	}
+
+	if grpcRequest.hostport == "" || grpcRequest.ServiceMethod == "" {
+		return fiber.NewErrorResponse(
+			fiberError.FiberError{
+				Code:    int(codes.InvalidArgument),
+				Message: "missing hostport/servicemethod",
+			})
+	}
+
+	conn, err := grpc.Dial(grpcRequest.hostport, grpc.WithTransportCredentials(insecure.NewCredentials()))
```

Comment:
grpc.Dial will create new connection to the target service. This should be cached as part of Dispatcher's field and reused by all calls

Reply:
Also, how can user pass additional configuration to the grpc client connection creation? Shall it be exposed as fiber config?

Reply:
> grpc.Dial will create new connection to the target service. This should be cached as part of Dispatcher's field and reused by all calls

Yes, initially I placed hostport/endpoint on the dispatcher. I took out simply because it can be isolated from the dispatcher. My thoughts were if there were multiple endpoint, then multiple dispatcher has to be created. 

The http dispatcher doesn't have this problem as there is no "dialing"  and the http `URL` is part of the request as well.

But I do agree that creating a new connection might be a bigger problem and we can fix this to the dispatcher.

> Also, how can user pass additional configuration to the grpc client connection creation? Shall it be exposed as fiber config?

Fair point i didnt think much about this, I can expose `DialOption`. Have a constructor that takes in dial option and hostport/endpoint that create a `conn`  that will be reused by all calls